src/compile/scale/range.ts

```diff
@@ -219,8 +225,16 @@ export function defaultRange(
       if (scaleType === 'ordinal') {
         // Only nominal data uses ordinal scale by default
         return type === 'nominal' ? 'category' : 'ordinal';
+      } else if (isContinuousToDiscrete(scaleType)) {
+        if (config.range && config.range.ordinal) {
```

Comment:
Don't forget to exclude this condition as the property could be specified in Vega config and be undefined in VL.